crates/core/tedge_api/src/workflow/supervisor.rs

```diff
@@ -229,6 +298,116 @@ impl WorkflowSupervisor {
     }
 }
 
+/// A stack of known versions for a workflow
+///
+/// In practice, one might have 3 concurrent versions:
+///
+/// - The current version, i.e. the version to be used for new operation instances.
+/// - One in-use version, that was used by some operation instance when the current version has been updated.
+/// - The builtin version provided by the agent and which can be overridden by the users.
```

Comment:
Since we don't allow multiple operations of the same type to run concurrently, I don't think this can happen.

Reply:
> we don't allow multiple operations of the same type to run concurrently

Yes, this is the assumption on which this work is based upon.

But, I'm afraid that @Bravo555 raised a valid concern.

- For the builtin operations, the commands are processed one after the other and the assumption is correct.
- But for user-defined operations, there is no mechanism to block a new command till the previous is done.
- A command instance topic cannot be re-used, but the two instances can run concurrently for the same command.
- => I will add a system test for @Bravo555 scenario.
- => I will have to fix the workflow engine to prevent concurrent runs for a given command.